btrfs-errmail.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+#
+# Copyright (c) 2022 Matthias Klumpp <matthias@tenstral.net>
+
+umask 022
+PATH=/sbin:/bin:/usr/sbin:/usr/bin
+export PATH
+
+if [ -f /etc/sysconfig/btrfsmaintenance ] ; then
+    . /etc/sysconfig/btrfsmaintenance
+fi
+
+if [ -f /etc/default/btrfsmaintenance ] ; then
+    . /etc/default/btrfsmaintenance
+fi
+
+. $(dirname $(realpath "$0"))/btrfsmaintenance-functions
+
+if [ -z "$BTRFS_MAILADDR" ]
+then
+      # no email set, nothing to do for us
+      exit 0
+fi
+
+BTRFS_STATS_MOUNTPOINTS=$(expand_auto_mountpoint "auto")
+OIFS="$IFS"
+IFS=:
+for MM in $BTRFS_STATS_MOUNTPOINTS; do
+	if ! is_btrfs "$MM"; then
+		echo "Path $MM is not btrfs, skipping"
+		continue
+	fi
+	devstats=$(btrfs device stats --check $MM 2>&1)
+	if [ $? -ne 0 ]; then
+		mail_body="$(sendmail -t <<EOF
```

Comment:
What if sendmail is not installed?

Reply:
Sendmail would obviously have to be a dependency of this. I changed the code so in case an email location was set but sendmail wasn't installed, the script will fail and print a warning to stderr.